kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableFeaturesSuite.scala

```diff
@@ -317,6 +317,68 @@ class DeltaTableFeaturesSuite extends DeltaTableWriteSuiteBase {
     }
   }
 
+  /* ---- Start: type widening tests ---- */
+  test("only typeWidening feature is enabled when metadata supports it: new table") {
```

Comment:
is there a way to create tests like this for variant (even though kernel by itself can't write/read variants)?

Reply:
currently we can't because we fail in the starting of the write itself.